src/Relude/List.hs

```diff
@@ -13,7 +13,34 @@ Utility functions to work with lists.
 module Relude.List
        ( module Relude.List.NonEmpty
        , module Relude.List.Reexport
+       , (!!?)
        ) where
 
+import Data.Bool (otherwise, (||))
+import Data.List (length, (!!))
+import Data.Maybe (Maybe (..))
+
+import Relude.Base ((<), (>=))
 import Relude.List.NonEmpty
 import Relude.List.Reexport
+import Relude.Numeric (Int)
+
+-- $setup
+-- >>> import Relude
+{- | Safer version of '!!', returns a Maybe.
```

Comment:
I believe this will be a Haddock error since `!!` is not in scope. Let's fix this as well :slightly_smiling_face: 
```suggestion
{- | Safer version of 'Relude.Unsafe.!!', returns a Maybe.
```